src/library/scala/collection/immutable/Vector.scala

```diff
@@ -1559,6 +1566,108 @@ final class VectorBuilder[A] extends ReusableBuilder[A, Vector[A]] {
     this
   }
 
+  //TODO Make public; this method is only private for binary compatibility
+  private[collection] def alignTo(before: Int, bigVector: Vector[A]): this.type = {
+    if (len1 != 0 || lenRest != 0)
+      throw new UnsupportedOperationException("A non-empty VectorBuilder cannot be aligned retrospectively. Please call .reset() or use a new VectorBuilder.")
+    val (prefixLength, maxPrefixLength) = bigVector match {
+      case Vector0 => (0, 1)
+      case v1: Vector1[_] => (0, 1)
+      case v2: Vector2[_] => (v2.len1, WIDTH)
+      case v3: Vector3[_] => (v3.len12, WIDTH2)
+      case v4: Vector4[_] => (v4.len123, WIDTH3)
+      case v5: Vector5[_] => (v5.len1234, WIDTH4)
+      case v6: Vector6[_] => (v6.len12345, WIDTH5)
+    }
+    if (maxPrefixLength == 1) return this // does not really make sense to align for <= 32 element-vector
+    val overallPrefixLength = (before + prefixLength) % maxPrefixLength
+    offset = (maxPrefixLength - overallPrefixLength) % maxPrefixLength
+    // pretend there are already `offset` elements added
+    advanceN(offset & ~MASK)
+    len1 = offset & MASK
+    prefixIsRightAligned = true
+    this
+  }
+
+  /**
+   * Removes `offset` leading `null`s in the prefix.
+   * This is needed after calling `alignTo` and subsequent additions,
+   * directly before the result is used for creating a new Vector.
+   *
+   * example:
+   *     a2 = Array(null, ..., null, Array(null, .., null, 0, 1, .., x), Array(x+1, .., x+32), ...)
+   * becomes
+   *     a2 = Array(Array(0, 1, .., x), Array(x+1, .., x+32), ...)
+   */
+  private[this] def leftAlignPrefix(): Unit = {
+    var a: Array[AnyRef] = null // the array we modify
+    var aParent: Array[AnyRef] = null // a's parent, so aParent(0) == a
+    if (depth >= 6) {
+      a = a6.asInstanceOf[Array[AnyRef]]
+      val i = offset >>> BITS5
+      if (i > 0) {
+        a = copyOfRange(a, i, LASTWIDTH)
```

Comment:
Note to myself: `copyOfRange` breaks VB, because after `result()` `aN.length` may be `< WIDTH` which is illegal